src/Middleware/WebSockets/test/UnitTests/AddWebSocketsTests.cs

```diff
@@ -17,14 +17,22 @@ public void AddWebSocketsConfiguresOptions()
         serviceCollection.AddWebSockets(o =>
         {
             o.KeepAliveInterval = TimeSpan.FromSeconds(1000);
+            o.KeepAliveTimeout = TimeSpan.FromSeconds(1234);
             o.AllowedOrigins.Add("someString");
         });
 
         var services = serviceCollection.BuildServiceProvider();
         var socketOptions = services.GetRequiredService<IOptions<WebSocketOptions>>().Value;
 
         Assert.Equal(TimeSpan.FromSeconds(1000), socketOptions.KeepAliveInterval);
+        Assert.Equal(TimeSpan.FromSeconds(1234), socketOptions.KeepAliveTimeout);
         Assert.Single(socketOptions.AllowedOrigins);
         Assert.Equal("someString", socketOptions.AllowedOrigins[0]);
     }
+
+    [Fact]
+    public void ThrowsForBadOptions()
+    {
+        Assert.Throws<ArgumentOutOfRangeException>(() => new WebSocketOptions() { KeepAliveTimeout = TimeSpan.FromMicroseconds(-1) });
```

Comment:
Can we add a test showing `Timeout.InfiniteTimeSpan` and `TimeSpan.Zero` are allowed and both disable the timeout?

Reply:
Idk how to test that they disable the timeout though. That's internal to the websocket impl, we'd need to use HttpClient to receive messages and then write a WebSocket frame parser to inspect the messages. And testing for a lack of Ping message from the server means just waiting for some arbitrary amount of time to check if we didn't get a message.